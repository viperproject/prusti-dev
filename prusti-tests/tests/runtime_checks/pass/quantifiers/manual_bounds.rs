//@run
use prusti_contracts::*;

#[insert_runtime_check]
#[requires(
    forall(
        |x: usize| (x < 20) ==> true
    )
)]
fn foo() {}

#[insert_runtime_check]
#[requires(
    forall(
        #[runtime_quantifier_bounds(0..20, 12..=22)]
        |x: usize, y: usize| (x < 20) && (y <= 22) && y >= 12 ==> x + y <= 41
    )
)]
fn bar() {}

#[trusted]
fn main() {
    foo();
    bar();
    println!("program intact");
}
