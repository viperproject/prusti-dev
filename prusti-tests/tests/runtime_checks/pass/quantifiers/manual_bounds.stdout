check function prusti_pre_check_item_foo is performed
check function prusti_pre_check_item_bar is performed
program intact
